You need to manually patch this into build/src-node for a local
node build of v0.8 *on your Mac*, if you do that.

From 532f9ffca2d9ac01e92a15a46adc0374ef75f4c4 Mon Sep 17 00:00:00 2001
From: Fedor Indutny <fedor.indutny@gmail.com>
Date: Thu, 31 Oct 2013 15:08:17 +0400
Subject: [PATCH] v8: backport b3775cf3 from upstream

---
 deps/v8/src/spaces.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/deps/v8/src/spaces.h b/deps/v8/src/spaces.h
index b0ecc5d..d76d77d 100644
--- a/deps/v8/src/spaces.h
+++ b/deps/v8/src/spaces.h
@@ -321,7 +321,8 @@ class MemoryChunk {
   Space* owner() const {
     if ((reinterpret_cast<intptr_t>(owner_) & kFailureTagMask) ==
         kFailureTag) {
-      return reinterpret_cast<Space*>(owner_ - kFailureTag);
+      return reinterpret_cast<Space*>(reinterpret_cast<intptr_t>(owner_) -
+                                      kFailureTag);
     } else {
       return NULL;
     }
-- 
1.8.5.1

